pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/CustomSerDeUtils.java

```diff
@@ -243,9 +243,14 @@ public TDigest deserialize(ByteBuffer byteBuffer) {
 
     @Override
     public byte[] serialize(Sketch value) {
-      // NOTE: Compact the sketch in unsorted, on-heap fashion for performance concern.
-      //       See https://datasketches.apache.org/docs/Theta/ThetaSize.html for more details.
-      return value.compact(false, null).toByteArray();
+      // The serializer should respect existing ordering to enable "early stop"
+      // optimisations on unions.
+      boolean shouldCompact = !value.isCompact();
+      boolean shouldOrder = value.isOrdered();
+      if (shouldCompact) {
```

Comment:
No - this is not the case and it depends on how the ingested sketches have been processed by the end user.  It might be the case that this is true for the empty and singleton sketch (sketch of a single hashed item), but not for others.